src/dials/command_line/index.py

```diff
@@ -167,6 +168,12 @@ def index(experiments, reflections, params):
     if params.indexing.image_range:
         reflections = slice_reflections(reflections, params.indexing.image_range)
 
+    if params.indexing.joint_indexing is Auto:
+        if any(e.is_still() for e in experiments):
```

Comment:
`any` or `all`?

Assert: if `any` but `not all` and `Auto` throw an error as expected behaviour not well defined here e.g. if someone tries to index a grid and rotation scan together (which I could imagine doing)